p4-16/psa/PSA.mdk

```diff
@@ -146,49 +143,105 @@ of native and non-native operations. Target documentation should highlight such
 implementations, so that application programmers are aware of the
 potentially significant performance penalties.
 
-Semantics of behavior for multiple calls to PRE APIs
+The pseudocode below defines where copies of packets will be made
+after the Ingress control block has completed executing, based upon
+the contents of several metadata fields in the struct
+`psa_ingress_output_metadata_t`.
+
+```
+[INCLUDE=psa.p4:Metadata_ingress_output]
+
+    psa_ingress_output_metadata_t ostd;
 
-The semantics of calling the PRE APIs is equivalent to setting
-intrinsic metadata fields/bits and assuming that the PRE looks up the
-fields in the following order: drop, truncate, multicast, clone,
-output_port.
+    if (ostd.clone) {
+        create a copy of the packet to the clone target
+        // TBD: Need a way to specify one (or more?) among multiple
+        // clone targets, if more than one such target exists.  Also
+        // to specify if anything is different about the cloned packet
+        // vs. other copies that might be made below.
+    }
+    // Continue below, regardless of whether a clone was created or not.
+    if (ostd.drop) {
+        drop the packet
+        return;   // Do not continue below.
+    }
+    if (ostd.resubmit) {
+        resubmit the packet, i.e. it will go back to starting with the
+            ingress parser;
+        // TBD: Specify if anything is different about the resubmitted
```

Comment:
My understanding is the resubmitted packet is coming into the ingress pipeline from the same port. There is no need to set the `egress_port` on the original packet because it is resubmitted for processing.